internal/server/data/provideruser.go

```diff
@@ -128,6 +130,14 @@ func ListProviderUsers(tx ReadTxn, opts ListProviderUsersOptions) ([]models.Prov
 	if opts.HideInactive {
 		query.B("AND active = ?", opts.HideInactive)
 	}
+	if opts.SCIMParameters != nil && opts.SCIMParameters.Filter != nil {
+		query.B("AND (")
+		err := filterSQL(*opts.SCIMParameters.Filter, query)
```

Comment:
```suggestion
		err := filterSQL(opts.SCIMParameters.Filter, query)
```

If we make this not a pointer we can remove the dereference here as well.